### PR TITLE
Remove unused FSDP1 components

### PR DESCRIPTION
**Pull request by krammnic**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [X] Clean up

Please link to any issues this PR addresses.

#### Changelog
What are the changes made in this PR?
* #1897

#### Test plan
Please make sure to do each of the following if applicable to your PR. If you're unsure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.

- [X] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [X] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [X] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Here is a [docstring example](https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285)
and a [tutorial example](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models)

- [X] I did not change any public API
- [ ] I have added an example to docs or docstrings

### CODE REVIEW

**Comment by krammnic** [comment]
Probably need to remove relations to docs

**Comment by ebsmothers** [comment]
Thanks for the PR! Looking at the code changes now, but at least for the doc build job failure you will need to remove [this line](https://github.com/pytorch/torchtune/blob/f560cbb21567ebf4852267681f7202526fa249ca/docs/source/api_ref_modules.rst?plain=1#L79) (you can check other rst files the same folder for any other APIs you've removed, their references will need to be removed as well)

File: torchtune/modules/peft/_utils.py [hunk 1]
```diff
@@ -318,6 +233,9 @@ def validate_missing_and_unexpected_for_lora(
     base_unexpected: Optional[List[str]] = None,
     lora_missing: Optional[List[str]] = None,
     lora_unexpected: Optional[List[str]] = None,
+    full_model_state_dict_keys: Optional[List[str]] = None,
+    lora_state_dict_keys: Optional[List[str]] = None,
+    base_model_state_dict_keys: Optional[List[str]] = None,
```
Comment (ebsmothers) [inline torchtune/modules/peft/_utils.py:238]: We should be able to do it without these extra arguments. You should be able to leverage base_missing, base_unexpected, lora_missing, lora_unexpected in the DPO recipe to use this API as is (may just need to move the call to `validate_missing_and_unexpected_for_lora` down a few lines)

File: torchtune/modules/peft/_utils.py [hunk 2]
```diff
@@ -362,19 +286,62 @@ def validate_missing_and_unexpected_for_lora(
         ]
     )
 
+    if full_model_state_dict_keys:
```
Comment (ebsmothers) [inline torchtune/modules/peft/_utils.py:289]: Same with all the code from here on down.. the changes should be doable without any modifications to this function

File: torchtune/training/_distributed.py [hunk 3]
```diff
@@ -22,46 +22,16 @@
...
-from torchtune.modules.peft import DoRALinear, LoRALinear
-from torchtune.modules.peft.lora import _lora_a_init_params, _lora_b_init_params
 from torchtune.utils import get_logger
 
 from torchtune.utils._device import get_device
 
 _log: logging.Logger = get_logger()
 
 
-FSDPPolicyType: Type = Callable[[nn.Module, bool, int], bool]
```
Comment (SalmanMohammadi) [inline torchtune/training/_distributed.py:38]: I think you need to remove the reference to this in `api_ref_training.rst` for the docs to build.

File: torchtune/training/__init__.py [hunk 4]
```diff
@@ -113,12 +108,7 @@
     "set_torch_num_threads",
     "shard_model",
     "get_shard_conditions",
-    "prepare_model_for_fsdp_with_meta_device",
     "validate_no_params_on_meta_device",
-    "contains_fsdp",
-    "FSDPPolicyType",
-    "get_full_finetune_fsdp_wrap_policy",
```
Comment (SalmanMohammadi) [inline torchtune/training/__init__.py:120]: Similar to my comment above, I think this will also need to be removed from the `torchtune.training` API ref doc.

Comment (SalmanMohammadi) [reply torchtune/training/__init__.py:120]: There's also a reference to this in the QAT recipe [in the note just below this heading](https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#qat-finetuning-recipe-in-torchtune). Should this note just be removed?

File: torchtune/training/__init__.py [hunk 5]
```diff
@@ -113,12 +108,7 @@
...
-    "lora_fsdp_wrap_policy",
```
Comment (SalmanMohammadi) [inline torchtune/training/__init__.py:121]: Same here to be removed from the training API docs.

Comment (ebsmothers) [reply torchtune/training/__init__.py:121]: Yeah that would be good to remove as well

File: torchtune/training/__init__.py (hunk 4 above)
Comment (ebsmothers) [reply torchtune/training/__init__.py:120]: Sorry started writing out a comment to this but I guess I never hit send. Yes we should remove the note about `memory_efficient_fsdp_wrap`. You can leave the field in any QAT configs as that'll get handled in #1854 but you can remove from e.g. [here](https://github.com/pytorch/torchtune/blob/d37c6aae9f1da5d1cdebfde02ac846c57b7677d7/recipes/configs/dev/8B_full_experimental.yaml#L71).

**Comment by krammnic** [comment]
Probably fine now

**Comment by ebsmothers** [comment]
Looks like CI is red?

**Comment by SalmanMohammadi** [comment]
I think you also need to remove the reference to `validate_state_dict_for_lora` in the API docs @krammnic

**Comment by SalmanMohammadi** [comment]
Hey @krammnic - think there's some merge conflicts to fix before I can kick off CI again.

**Comment by ebsmothers** [comment]
@krammnic I’m gonna close this now that #2016 has landed. Thanks for your work on this one!
